pkg/reconciler/taskrun/validate_resources.go

```diff
@@ -445,119 +459,226 @@ func extractParamIndex(paramReference string, arrayParams map[string]int, outofB
 			}
 		}
 		if v.VolumeSource.Secret != nil {
-			extractParamIndex(v.Secret.SecretName, arrayParams, outofBoundParams)
+			err := FindInvalidParamArrayReference(ctx, v.Secret.SecretName, arrayParams, outofBoundParams)
+			if err != nil {
+				return err
+			}
 			for _, item := range v.Secret.Items {
-				extractParamIndex(item.Key, arrayParams, outofBoundParams)
-				extractParamIndex(item.Path, arrayParams, outofBoundParams)
+				err := FindInvalidParamArrayReference(ctx, item.Key, arrayParams, outofBoundParams)
+				if err != nil {
+					return err
+				}
+				err = FindInvalidParamArrayReference(ctx, item.Path, arrayParams, outofBoundParams)
+				if err != nil {
+					return err
+				}
 			}
 		}
 		if v.PersistentVolumeClaim != nil {
-			extractParamIndex(v.PersistentVolumeClaim.ClaimName, arrayParams, outofBoundParams)
+			err := FindInvalidParamArrayReference(ctx, v.PersistentVolumeClaim.ClaimName, arrayParams, outofBoundParams)
+			if err != nil {
+				return err
+			}
 		}
 		if v.Projected != nil {
 			for _, s := range volumes[i].Projected.Sources {
 				if s.ConfigMap != nil {
-					extractParamIndex(s.ConfigMap.Name, arrayParams, outofBoundParams)
+					err := FindInvalidParamArrayReference(ctx, s.ConfigMap.Name, arrayParams, outofBoundParams)
+					if err != nil {
+						return err
+					}
 				}
 				if s.Secret != nil {
-					extractParamIndex(s.Secret.Name, arrayParams, outofBoundParams)
+					err := FindInvalidParamArrayReference(ctx, s.Secret.Name, arrayParams, outofBoundParams)
+					if err != nil {
+						return err
+					}
 				}
 				if s.ServiceAccountToken != nil {
-					extractParamIndex(s.ServiceAccountToken.Audience, arrayParams, outofBoundParams)
+					err := FindInvalidParamArrayReference(ctx, s.ServiceAccountToken.Audience, arrayParams, outofBoundParams)
+					if err != nil {
+						return err
+					}
 				}
 			}
 		}
 		if v.CSI != nil {
 			if v.CSI.NodePublishSecretRef != nil {
-				extractParamIndex(v.CSI.NodePublishSecretRef.Name, arrayParams, outofBoundParams)
+				err := FindInvalidParamArrayReference(ctx, v.CSI.NodePublishSecretRef.Name, arrayParams, outofBoundParams)
+				if err != nil {
+					return err
+				}
 			}
 			if v.CSI.VolumeAttributes != nil {
 				for _, value := range v.CSI.VolumeAttributes {
-					extractParamIndex(value, arrayParams, outofBoundParams)
+					err := FindInvalidParamArrayReference(ctx, value, arrayParams, outofBoundParams)
+					if err != nil {
+						return err
+					}
 				}
 			}
 		}
 	}
+	return nil
 }
 
-func validateContainerParamArrayIndexing(c *corev1.Container, arrayParams map[string]int, outofBoundParams *sets.String) {
-	extractParamIndex(c.Name, arrayParams, outofBoundParams)
-	extractParamIndex(c.Image, arrayParams, outofBoundParams)
-	extractParamIndex(string(c.ImagePullPolicy), arrayParams, outofBoundParams)
+func validateContainerParamArrayIndexing(ctx context.Context, c *corev1.Container, arrayParams map[string]int, outofBoundParams *sets.String) error {
```

Comment:
NIT: please add docstring, this is not needed since its a private function but will be helpful for the future contributors. thanks!